File: Test/controllers/regist.go
package controllers

import (
	"github.com/astaxie/beego"
	_"github.com/go-sql-driver/mysql"
	"fmt"
	"github.com/astaxie/beego/orm"
	"Test/models"
	"crypto/md5"
	"encoding/hex"
	"time"
)

type RegistController struct {
	beego.Controller
}


func (this *RegistController) Get(){
	this.TplName = "regist.html"
}

func (this *RegistController) Post(){
	//获取用户名
	username := this.Input().Get("username")
	fmt.Println(username)
	//获取密码
	password := this.Input().Get("password")
	if username == "" || password == ""{
		fmt.Println("账号密码不能为空！")
		this.TplName = "regist.html"
		return
	}
	//存储账号密码
	o := orm.NewOrm()
	user := new(models.User)
	user.Username = username
	hash := md5.New()
	hash.Write([]byte(password))
	//密码加密处理
	user.Password = hex.EncodeToString(hash.Sum(nil))
	user.Addtime = int(time.Now().Unix())
	//进行插入操作
	n,_ := o.Insert(user)
	if n>0{
		//插入成功
		fmt.Println(user)
		this.Ctx.WriteString("suc")
		fmt.Println("注册成功！")
		this.Redirect("/login",302)
	}else{
		fmt.Println("注册失败")
	}

	//this.TplName = "login.html"
}

File: Test/static/js/regist.js
$("#regist").click(function() {
	// alert("login")
	//获取用户名
	var username = $("#username").val();
	//获取用户密码
	var password = $("#password").val();
	//获取其他密码
	var repeatpass = $("#repeatpass").val();
	//进行比较
	if(password !== repeatpass){
        var mymessage=confirm("两次输入密码不一致，是否重新注册？");
        if(mymessage==true)
        {
            $("#username").val("")
            $("#password").val("")
            $("#repeatpass").val("")
        }
    }else{
		//提交进行相关验证
		$.post("/regist",{username:username,password:password},function(data){
			if(data == "suc"){
			    //跳转到登录也
                window.location.href = "/login"
            }
		})
	}
})
